src/hotspot/os/posix/os_posix.cpp

```diff
@@ -711,7 +711,16 @@ void* os::get_default_process_handle() {
 }
 
 void* os::dll_lookup(void* handle, const char* name) {
-  return dlsym(handle, name);
+  ::dlerror(); // Clear any previous error
+  void* ret = ::dlsym(handle, name);
+  if (ret == nullptr) {
+    const char* tmp = ::dlerror();
+    // It is possible that we found a NULL symbol, hence no error.
+    if (tmp != nullptr) {
+      log_debug(os)("Symbol %s not found in dll: %s", name, tmp);
```

Comment:
I'm still a little confused about the message here. Are we just reporting that the symbol doesn't exist, or are we actually reporting that the attempt to lookup the symbol failed due to some error?

Reply:
Yes, the issue is that we don't know if there was an error or no such symbol, "undefined symbol".
From dlsym/dlerror API perspective "symbol doesn't exist" is an error.
But for our case it would be logical to separate them.

All we know is that the user tried to find X in dll-file, and it was not found due to "string".

Happy to change it, but I can't think of a better message. For my case with hsdis this would had been good enough.

Reply:
Okay